packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/validate-certification-request.handler.ts

```diff
@@ -0,0 +1,55 @@
+import * as Moment from 'moment';
+import { extendMoment } from 'moment-range';
+import { CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Not, Repository } from 'typeorm';
+import { ISuccessResponse, ResponseFailure, ResponseSuccess } from '@energyweb/origin-backend-core';
+import { HttpStatus } from '@nestjs/common';
+import {
+    ValidateCertificationRequestCommand,
+    CertificationRequestStatus
+} from '@energyweb/issuer-api';
+import { CertificationRequest } from '../certification-request.entity';
+
+@CommandHandler(ValidateCertificationRequestCommand)
+export class ValidateCertificationRequestHandler
+    implements ICommandHandler<ValidateCertificationRequestCommand>
+{
+    constructor(
+        @InjectRepository(CertificationRequest)
+        private readonly repository: Repository<CertificationRequest>
+    ) {}
+
+    async execute({
+        dto: { fromTime, toTime, deviceId }
+    }: ValidateCertificationRequestCommand): Promise<ISuccessResponse> {
+        const moment = extendMoment(Moment);
+        const unix = (timestamp: number) => moment.unix(timestamp);
+
+        const deviceCertificationRequests = await this.repository.find({
+            where: {
+                revoked: false,
+                deviceId,
+                status: Not(CertificationRequestStatus.Error)
+            }
+        });
+
+        const generationTimeRange = moment.range(unix(fromTime), unix(toTime));
+
+        for (const certificationRequest of deviceCertificationRequests) {
+            const certificationRequestGenerationRange = moment.range(
+                unix(certificationRequest.fromTime),
+                unix(certificationRequest.toTime)
+            );
+
+            if (generationTimeRange.overlaps(certificationRequestGenerationRange)) {
+                return ResponseFailure(
+                    `Wanted generation time clashes with an existing certification request: ${certificationRequest.id}`,
+                    HttpStatus.CONFLICT
+                );
+            }
+        }
+
+        return ResponseSuccess('The generation period is valid.');
+    }
+}
```

Comment:
What's the difference between this handler and the one in the issuer-api?
https://github.com/energywebfoundation/origin/blob/121fafd84515d482b42b477fe9f8c617481d22b2/packages/traceability/issuer-api/src/pods/certification-request/handlers/validate-certification-request.handler.ts

Reply:
same as above